### PR TITLE
rosdep: pylint3 unavailable on Fedora, Ubuntu 20.04 onward

### PR DESCRIPTION
**Pull request by 130s**
# Issue

- `rosdep install`  exec for `python-pylint3` rosdep key error on Jammy (but not on <s>Focal</s> Bionic. Later [I confirmed it fails on Focal](https://github.com/ros/rosdistro/actions/runs/5197879128)).
   ```
   # lsb_release -a; apt update; rosdep update; rosdep install --from-paths src --ignore-src -r -y
   :
   Description:	Ubuntu 22.04.2 LTS
   :
   executing command [apt-get install -y pylint3]
   Reading package lists... Done
   Building dependency tree... Done
   Reading state information... Done
   E: Unable to locate package pylint3
   ERROR: the following rosdeps failed to install
     apt: command [apt-get install -y pylint3] failed
     apt: Failed to detect successful installation of [pylint3]
   ```
   I interpreted the output above as "The requested rosdep key does exist, but apt pkg doesn't".

   `apt` fails too.
   ```
   # apt install pylint3
   Reading package lists... Done
   Building dependency tree... Done
   Reading state information... Done
   E: Unable to locate package pylint3
   ```
# Background
- The rosdep key seems to have been added in https://github.com/ros/rosdistro/pull/21204, which includes websites as rationale.
   - Same web links still exist.
      - Debian https://packages.debian.org/bullseye/pylint3
      - gentoo https://packages.gentoo.org/packages/dev-python/pylint
   - Same links lead to "Not found" or equivalent.
      - Fedora https://apps.fedoraproject.org/packages/python3-pylint
      - Ubuntu website https://packages.ubuntu.com/bionic/pylint3 
- On Bionic (Docker image that I just pulled. Haven't verified when it was built) I confirmed the pkg is still available via `apt`.
   <details><summary>log</summary>
   
   ```
   $ docker pull ros:melodic-ros-base-bionic 
   :
   36b05b7b63adec64f29ed7b397779f11123dcfff3b262a90af818a785
   Status: Downloaded newer image for ros:melodic-ros-base-bionic
   docker.io/library/ros:melodic-ros-base-bionic
   $ docker run -it ros:melodic-ros-base-bionic bash
   
   root@44b2cff94d2a:/# lsb_release -a
   No LSB modules are available.
   Distributor ID: Ubuntu
   Description:    Ubuntu 18.04.6 LTS
   Release:        18.04
   
   root@44b2cff94d2a:/# apt update; apt install pylint3
   :
   Get:20 http://archive.ubuntu.com/ubuntu bionic-backports/main amd64 Packages [64.0 kB]              Fetched 28.9 MB in 1min 12s (399 kB/s)
   :
   Reading state information... Done
   The following additional packages will be installed:
     blt libtcl8.6 libtk8.6 libxft2 libxss1 python3-astroid python3-isort python3-lazy-object-proxy python3-logilab-common python3-mccabe python3-pkg-resources python3-setuptools python3-six python3-tk python3-wrapt tk8.6-blt2.5
   Suggested packages:
     blt-demo tcl8.6 tk8.6 pylint-doc python-setuptools-doc tix python3-tk-dbg
   The following NEW packages will be installed:
     blt libtcl8.6 libtk8.6 libxft2 libxss1 pylint3 python3-astroid python3-isort python3-lazy-object-proxy python3-logilab-common python3-mccabe python3-pkg-resources python3-setuptools python3-six python3-tk python3-wrapt tk8.6-blt2.5
   0 upgraded, 17 newly installed, 0 to remove and 0 not upgraded.
   Need to get 3304 kB of archives.
   After this operation, 16.0 MB of additional disk space will be used.
   Do you want to continue? [Y/n] n
   Abort.
   ```
   </details>
# Open question
- [x] I only assume this pkg will no longer be added back to future distros. I was not sure how to describe that in the rosdep key list. --> Because it's only available on Bionic on Ubuntu distros that are reasonably in use, set it only to Bionic.
- [x] Haven't verified on running Fedora. --> https://github.com/ros/rosdistro/pull/37509#discussion_r1224641530 Fedora ver is confirmed to exist.

### CODE REVIEW

File: rosdep/python.yaml
Comment (sloretz) [inline rosdep/python.yaml:3101]: It looks like `pylint3` was renamed to `pylint`. I think we should make this `pylint` instead of `null` since that is its name in Focal and beyond..

* `pylint3` was transitioned with a dummy package with a dependency on new name in Debian: https://packages.debian.org/bullseye/pylint3
* pylint going forward in Debian: https://packages.debian.org/sid/pylint
* Ubuntu link: https://packages.ubuntu.com/focal/pylint

**Review by sloretz — DISMISSED** [review]

File: rosdep/python.yaml
Comment (emersonknapp) [reply rosdep/python.yaml:3101]: Beat me to it - I was just combing thru ubuntu package listings :) I can second that this is also the conclusion I came to

File: rosdep/python.yaml
Comment (emersonknapp) [inline rosdep/python.yaml:3099]: Fedora does provide this package https://packages.fedoraproject.org/pkgs/pylint/python3-pylint/ - I was able to just `docker run -it fedora` and then `yum install python3-pylint` with no issues - what fedora version did you test this on to rule it out?

Comment (130s) [reply rosdep/python.yaml:3099]: Thanks for checking. No, as I mentioned in OP, I had not tested on Fedora.

Update pushed.

**Review by emersonknapp — APPROVED** [review]

**Comment by 130s** [comment]
@ros/rosdeputies What's blocking the review of this?

Also I see some other unmerged PRs for dep with green light, while some dep PRs are getting merged. What's the timeline in general for the review?

**Comment by emersonknapp** [comment]
>  What's blocking the review of this?

It's a little bit randomized, we try to keep on top of everything but as it is manual with a weekly rotation of responsibilities, things sometimes slip.

The only remaining thing here is that new rosdep keys require two approvers - but there shouldn't be any concerns blocking that. 
@sloretz - or @adityapande-1995 @mjcarroll on distro duty this week - can we get a second approval here for merge?

**Review by mjcarroll — APPROVED** [review]

**Pull request merged**
